test/Grpc.Net.Client.Tests/AsyncUnaryCallTests.cs

```diff
@@ -76,14 +77,18 @@ public async Task AsyncUnaryCall_Success_HttpRequestMessagePopulated()
             Assert.AreEqual("identity,gzip", httpRequestMessage.Headers.GetValues(GrpcProtocolConstants.MessageAcceptEncodingHeader).Single());
             Assert.AreEqual(null, requestContentLength);
 
-            var userAgent = httpRequestMessage.Headers.UserAgent.Single()!;
-            Assert.AreEqual("grpc-dotnet", userAgent.Product?.Name);
-            Assert.IsTrue(!string.IsNullOrEmpty(userAgent.Product?.Version));
+            var grpcVersion = httpRequestMessage.Headers.UserAgent.First();
```

Comment:
we probably need some tests for the user-agent functionality (e.g. at least check the format looks the way we expect it to look with a regex).

Reply:
Followed the recommendation below and added a separate test class.